Code/Framework/AzCore/AzCore/Math/Vector2.h

```diff
@@ -163,6 +163,19 @@ namespace AZ
         //! The positive rotation direction is defined such that the x-axis is rotated into the y-axis.
         Vector2 GetPerpendicular() const;
 
+       // carbonated begin enable_catbonated_1: Methods called from o2de-gruber
+#if defined(CARBONATED)
+#define AZ_MATH_FORCE_INLINE AZ_FORCE_INLINE
+        /**
+         * Rotates the vector r radians.
+         */
+        AZ_MATH_FORCE_INLINE const Vector2 Rotate(float r) const
+        {
+            return Vector2(cos(r) * m_x - sin(r) * m_y, sin(r) * m_x + cos(r) * m_y);
```

Comment:
We have a simd sin/cos implementation now inside AZ::Simd that would dramatically speed this up..

Reply:
https://github.com/o3de/o3de/blob/3c74a21e310dbc68d982a9de3cb095f73f727f60/Code/Framework/AzCore/AzCore/Math/MathUtils.h#L298